### PR TITLE
Feature/tpfs drop

### PR DESCRIPTION
**Pull request by anew**
Fix for the time partitioned file set to make sure that 
- the Hive table gets deleted when the dataset is dropped
- when a partition is dropped from the dataset, it is also removed from Hive
Also adds unit test that verifies these.

### CODE REVIEW

**Comment by anew** [comment]
Build succeeded here: https://builds.cask.co/chain/viewChain.action?planKey=CDAP-DUT591

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/HiveExploreServiceFileSetTest.java [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.text.DateFormat;
+import java.util.Collections;
+
+/**
+ *
```
Comment (gokulavasan) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/HiveExploreServiceFileSetTest.java]: doc?

**Comment by gokulavasan** [comment]
LGTM :+1:

File: cdap-explore/src/test/java/co/cask/cdap/explore/service/HiveExploreServiceFileSetTest.java (hunk 1 above)
Comment (anew) [inline cdap-explore/src/test/java/co/cask/cdap/explore/service/HiveExploreServiceFileSetTest.java]: I thought we don't require that for tests. But I can add a comment.

**Pull request merged**